Lib/shutil.py

```diff
@@ -42,6 +42,14 @@
 except ImportError:
     getgrnam = None
 
+if os.name == 'posix':
+    import posix
+elif os.name == 'nt':
+    import nt
+
+_HAS_SENDFILE = hasattr(os, "sendfile")
+_HAS_FCOPYFILE = hasattr(posix, "_fcopyfile")
```

Comment:
Spotted this just then - `posix` is not defined on Windows at this point.

Reply:
fixed, thanks